server/Arcadia.Assistant.Web/Controllers/CalendarEventsController.cs

```diff
@@ -162,7 +162,16 @@ public CalendarEventsController(ITimeoutSettings timeoutSettings, IEmployeesRegi
                 return this.NotFound();
             }
 
-            if (!(await this.authorizationService.AuthorizeAsync(this.User, employee, new ApproveCalendarEvents())).Succeeded)
+            var approvedStatus = new CalendarEventStatuses().ApprovedForType(model.Type);
```

Comment:
can this logic be moved on Requirement level?

Reply:
also, this probably should be 'EditCalendarEvents' and 'EditPendingCalendarEvents'. That way we also check date changes